src/Block/AdminPreviewBlockService.php

```diff
@@ -0,0 +1,135 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the Sonata Project package.
+ *
+ * (c) Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sonata\AdminBundle\Block;
+
+use Sonata\AdminBundle\Admin\AdminInterface;
+use Sonata\AdminBundle\Admin\Pool;
+use Sonata\AdminBundle\Datagrid\ListMapper;
+use Sonata\BlockBundle\Block\BlockContextInterface;
+use Sonata\BlockBundle\Block\Service\AbstractBlockService;
+use Symfony\Component\DependencyInjection\Exception\ServiceNotFoundException;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpFoundation\Session\Session;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+use Twig\Environment;
+
+/**
+ * @author Javier Spagnoletti <phansys@gmail.com>
+ */
+final class AdminPreviewBlockService extends AbstractBlockService
+{
+    /**
+     * @var Pool
+     */
+    private $pool;
+
+    public function __construct(Environment $twig, Pool $pool)
+    {
+        parent::__construct($twig);
+
+        $this->pool = $pool;
+    }
+
+    public function execute(BlockContextInterface $blockContext, ?Response $response = null): Response
+    {
+        $admin = $this->getAdmin($blockContext->getSetting('code'));
+        $this->handleFilters($admin, $blockContext);
+
+        foreach ($blockContext->getSetting('remove_list_fields') as $listField) {
+            $admin->getList()->remove($listField);
+        }
+
+        $datagrid = $admin->getDatagrid();
+
+        return $this->renderPrivateResponse($blockContext->getTemplate(), [
+            'block' => $blockContext->getBlock(),
+            'settings' => $blockContext->getSettings(),
+            'admin' => $admin,
+            'datagrid' => $datagrid,
+        ], $response);
+    }
+
+    public function getName()
+    {
+        return 'Admin preview';
+    }
+
+    public function configureSettings(OptionsResolver $resolver): void
+    {
+        $resolver->setDefaults([
+            'text' => 'Preview',
+            'filters' => [],
+            'icon' => false,
+            'limit' => 10,
+            'code' => false,
+            'template' => '@SonataAdmin/Block/block_admin_preview.html.twig',
+            'remove_list_fields' => [ListMapper::NAME_ACTIONS],
+        ]);
+    }
+
+    /**
+     * @throws \InvalidArgumentException if the provided admin code is invalid
+     */
+    private function getAdmin(string $code): AdminInterface
+    {
+        try {
+            $admin = $this->pool->getAdminByAdminCode($code);
+        } catch (ServiceNotFoundException $e) {
```

Comment:
I think `ServiceNotFoundException` is never thrown from `Pool::getAdminByAdminCode`, instead is thrown `AdminCodeNotFoundException` which extends from `\InvalidArgumentException`, so maybe there is no need of try-catch.

Reply:
You're right. Removed.